### PR TITLE
fix(FEC-11697): 2 captions are selected in the menu

### PR DESCRIPTION
**Pull request by JonathanTGold**
### Description of the Changes

see https://github.com/kaltura/playkit-js/pull/633

**in addition regardless of the original issue** 
add error handling on load() method.
**issue:**  the promise will remain 'pending' forever ... in case an manifest loading error
Since the promise resolving returned from the load() rely on Hlsjs.Events.MANIFEST_LOADED event https://github.com/kaltura/playkit-js-hls/blob/master/src/hls-adapter.js#L140 to be resolved which is not fired in case of en error
**fix:** call promise reject in case of fatal error and still pending promise

solves: FEC-11697 and FEC-11893

related pr: 
https://github.com/kaltura/playkit-js-dash/pull/182
https://github.com/kaltura/playkit-js/pull/633

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/hls-adapter.js [hunk 1]
```diff
@@ -710,11 +709,10 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
         active: CEATextTrack.mode === 'showing',
         label: CEATextTrack.label,
         kind: CEATextTrack.kind,
-        language: CEATextTrack.language,
-        index: this._playerTracks.filter(track => track instanceof TextTrack).length
+        language: CEATextTrack.language
       };
       textTrack = new TextTrack(settings);
-      this._nativeTextTracksMap[settings.index] = CEATextTrack;
+      this._nativeTextTracksMap[textTrack.index] = CEATextTrack;
```
Comment (yairans) [inline src/hls-adapter.js:715]: use map like in playkit?

Comment (JonathanTGold) [reply src/hls-adapter.js:715]: done

File: package.json [hunk 2]
```diff
@@ -102,7 +103,7 @@
     "webpack-dev-server": "^3.11.0"
   },
   "peerDependencies": {
-    "@playkit-js/playkit-js": "canary",
+    "@playkit-js/playkit-js": "kaltura/playkit-js#FEC-11697",
```
Comment (yairans) [inline package.json:106]: remove

File: src/hls-adapter.js [hunk 3]
```diff
@@ -1101,6 +1099,11 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
     const errorFatal = data.fatal;
     let errorDataObject = this._getErrorDataObject(data);
     if (errorFatal) {
+      if (this._loadPromiseHandlers) {
+        const {type, details, url} = data;
+        this._loadPromiseHandlers.reject({type, details, url});
```
Comment (yairans) [inline src/hls-adapter.js:1104]: reject after it has been resolved? sounds incorrect

Comment (yairans) [reply src/hls-adapter.js:1105]: do you mean which is NOT fired in case of an error?

File: src/hls-adapter.js [hunk 4]
```diff
@@ -539,8 +539,8 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
   load(startTime: ?number): Promise<Object> {
     if (!this._loadPromise) {
       this._startTime = startTime;
-      this._loadPromise = new Promise(resolve => {
-        this._resolveLoad = resolve;
+      this._loadPromise = new Promise((resolve, reject) => {
+        this._loadPromiseHandlers = {resolve, reject};
```
Comment (yairans) [inline src/hls-adapter.js:543]: declare it like the others and reset it where need

File: src/hls-adapter.js (hunk 3 above)
Comment (JonathanTGold) [reply src/hls-adapter.js:1105]: Right fixed

File: src/hls-adapter.js (hunk 4 above)
Comment (JonathanTGold) [reply src/hls-adapter.js:543]: need to be reset only on resolve and reject (2 places)

Comment (yairans) [reply src/hls-adapter.js:551]: in _reset and destroy also no?

Comment (JonathanTGold) [reply src/hls-adapter.js:551]: one of the two options (manifest loaded , manifest loading error) will always have to happen, no third option, and it is reset to null in both these cases, so i think any additional reset is redundant

File: src/hls-adapter.js [hunk 5]
```diff
@@ -600,6 +609,11 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
     this._responseFilterError = false;
     this._hls.detachMedia();
     this._hls.destroy();
+    if (this._loadPromiseHandlers) {
```
Comment (yairans) [inline src/hls-adapter.js:612]: momve to destroy. else `_reloadWithDirectManifest` will get rejected promise

File: src/hls-adapter.js [hunk 6]
```diff
@@ -1080,6 +1093,11 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
     const errorFatal = data.fatal;
     let errorDataObject = this._getErrorDataObject(data);
     if (errorFatal) {
+      if (this._loadPromiseHandlers) {
```
Comment (yairans) [inline src/hls-adapter.js:1096]: unnecessary. the promise will rejected by destroy if it's critical error. if not it should be pending

File: src/hls-adapter.js (hunk 5 above)
Comment (JonathanTGold) [reply src/hls-adapter.js]: done

File: src/hls-adapter.js (hunk 6 above)
Comment (JonathanTGold) [reply src/hls-adapter.js]: done

File: src/hls-adapter.js [hunk 7]
```diff
@@ -576,15 +588,17 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
       super.destroy().then(
         () => {
           HlsAdapter._logger.debug('destroy');
-          this._loadPromise = null;
           this._playerTracks = [];
-          this._nativeTextTracksMap = [];
+          this._nativeTextTracksMap = {};
           this._sameFragSNLoadedCount = 0;
           this._lastLoadedFragSN = -1;
+          this._loadPromiseHandlers?.reject('The adapter has been destroyed while loading');
```
[the remark this answers is not in the document]
Comment (JonathanTGold) [reply src/hls-adapter.js]: done

File: src/hls-adapter.js [hunk 8]
```diff
@@ -492,6 +500,8 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
         this._config.hlsConfig.startPosition = this.currentTime;
       }
       this._reset();
+      this._loadPromiseHandlers?.reject('media detached while loading');
```
[the remark this answers is not in the document]
Comment (JonathanTGold) [reply src/hls-adapter.js]: done

**Pull request merged**